src/Fmt_ast.ml

```diff
@@ -3514,15 +3514,29 @@ and fmt_structure c ctx itms =
           | _ -> false
         in
         let is_simple itm =
-          match itm.pstr_desc with
-          | Pstr_include {pincl_mod= me} | Pstr_module {pmb_expr= me} ->
+          match (c.conf.let_decl, itm.pstr_desc) with
+          | _, Pstr_include {pincl_mod= me} | _, Pstr_module {pmb_expr= me} ->
               is_simple_mod me
-          | Pstr_open _ -> true
+          | _, Pstr_open _ -> true
+          | `Compact, Pstr_value _ ->
+              itm.pstr_loc.loc_start.pos_lnum
+              = itm.pstr_loc.loc_end.pos_lnum
```

Comment:
In `Ast.is_simple` the width of formatting the expression without breaking is considered when determining whether an expression "is simple". How would you feel about using a similar approach here instead of using the input source locations? My concern is that if the input has a let binding on one line, but it is too wide to fit within the margin, then the surrounding spacing will be "wrong".

Another point that is not really specific to this change is that for any input file, we should aim for ocamlformat to have the robustness property that formatting the input with margin X and then with margin Y and then again with X should produce the same output as formatting just once with margin X.

Uses of the source locations such as done in this patch make me hesitant because they will lead to this property failing, with the result that formatting is less easily predictable and understandable.